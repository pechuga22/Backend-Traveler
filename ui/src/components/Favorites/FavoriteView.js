import React from "react";
import { Link } from "react-router-dom";
import { Container, Icon, Grid, Header, Message } from "semantic-ui-react";

import FavoritePack from "./FavoritePack";

export default function FavoriteView({ tours }) {
  return (
    <Container text className="favorites">
      <Grid columns={3} padded>
        <Grid.Column width={4}>
          <Link to={"/"}>
            <Icon color="pink" size="large" className="arrow left header" />
          </Link>
        </Grid.Column>
        <Grid.Column textAlign="center" width={8}>
          <Header color="pink">
            <h2>My Favorites</h2>
          </Header>
        </Grid.Column>
      </Grid>
      <Grid.Row>
        <Grid columns={2}>
          {tours.length >= 1 ? (
            tours &&
            tours.map((tour) => <FavoritePack tour={tour} key={tour._id} />)
          ) : (
            <Grid textAlign="center" padded>
              <Message color="yellow">Don't have favorite tours</Message>
            </Grid>
          )}
        </Grid>
      </Grid.Row>
    </Container>
  );
}
