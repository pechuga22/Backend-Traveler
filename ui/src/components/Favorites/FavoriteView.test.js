import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

import FavoriteView from './FavoriteView'

const tours = [
  {
    _id: '5ec573d6f235707135952a95',
    title: 'Camino a la piedra de Guatapé',
  },
  {
    _id: '5ec573d6f235707135952a96',
    title: 'Historia de las playas de Bahia a pie',
  },
  {
    _id: '5ec574f6f235707135952a97',
    title: 'Cuba en bus, conociendo la Revolución Cubana',
  },
  {
    _id: '5ec574f6f235707135952a98',
    title: 'Ruta del tintico, lo mejor del café de Colombia',
  }
]

describe('./components/Favorites/FavoriteView.js', () => {
  describe('.render()', () => {
    it('should render FavoriteView component', () => {
      const wrapper = shallow(
        <FavoriteView tours={tours} />
      )
      expect(toJson(wrapper)).toMatchSnapshot()
    })
    it('renders text FavoriteView', () => {
      const wrapper = shallow(
        <FavoriteView tours={tours} />
      )
      expect(wrapper.find('h2').text()).toBe('My Favorites')
    })
  })
})
