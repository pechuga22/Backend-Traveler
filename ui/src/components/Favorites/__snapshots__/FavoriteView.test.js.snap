// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`./components/Favorites/FavoriteView.js .render() should render FavoriteView component 1`] = `
<Container
  className="favorites"
  text={true}
>
  <Grid
    columns={3}
    padded={true}
  >
    <GridColumn
      width={4}
    >
      <Link
        to="/"
      >
        <Icon
          as="i"
          className="arrow left header"
          color="pink"
          size="large"
        />
      </Link>
    </GridColumn>
    <GridColumn
      textAlign="center"
      width={8}
    >
      <Header
        color="pink"
      >
        <h2>
          My Favorites
        </h2>
      </Header>
    </GridColumn>
  </Grid>
  <GridRow>
    <Grid
      columns={2}
    >
      <FavoritePack
        key="5ec573d6f235707135952a95"
        tour={
          Object {
            "_id": "5ec573d6f235707135952a95",
            "title": "Camino a la piedra de Guatapé",
          }
        }
      />
      <FavoritePack
        key="5ec573d6f235707135952a96"
        tour={
          Object {
            "_id": "5ec573d6f235707135952a96",
            "title": "Historia de las playas de Bahia a pie",
          }
        }
      />
      <FavoritePack
        key="5ec574f6f235707135952a97"
        tour={
          Object {
            "_id": "5ec574f6f235707135952a97",
            "title": "Cuba en bus, conociendo la Revolución Cubana",
          }
        }
      />
      <FavoritePack
        key="5ec574f6f235707135952a98"
        tour={
          Object {
            "_id": "5ec574f6f235707135952a98",
            "title": "Ruta del tintico, lo mejor del café de Colombia",
          }
        }
      />
    </Grid>
  </GridRow>
</Container>
`;
